### DISCUSSION TITLE
SecureFields re-rendering unexpectedly

### DISCUSSION THREAD

**Discussion by juanpernett1299** [mercadopago/sdk-react#69] [Q&A] [answered]
Hi. I'm trying to integrate secure fields with react. Im experiencing an unexpected behavior where the iframes reloads, and lose the input data, when one state in the page changes. This is quite weird beacause it also happens when the browser windows size changes. Previously i also tried to integrate PaymentBrick and had the same issue when resizing the browser window

**Comment by lemazzo** [comment, marked as answer]
Hello, @juanpernett1299

As security precaution, the expected behavior is **not** keeping the input data between renders.

The loss of input information can happen if you are allowing the component to be reloaded or have a state controlling the size of the window.

Even for Payment Brick, we advise you to review how the states were implemented in your application to ensure that re-renders do not occur and the component instance is not lost.

We hope it works out.

Thank you!
